docs/CONTRIBUTING.md

```diff
@@ -56,18 +56,41 @@ Vector team member will find this document useful.
 
 ### New sources, sinks, and transforms
 
-If you're contributing a new source, sink, or transform to Vector, thank you that's way cool! There's a few steps you
-need to think about if you want to make sure we can merge your contribution. We're here to help you along with these steps,
-but they are a blocker to getting a new integration released.
-
-To merge a new source, sink, or transform, you need to:
+If you're thinking of contributing a new source, sink, or transform to Vector, thank you that's way cool! There's a few
+steps you need to think about if you want to make sure we can merge your contribution. We're here to help you along with
+these steps, but they are a blocker to getting a new integration released.
+
+Prior to beginning work on a new source or sink if a GitHub Issue does not already exist, please open one to discuss the
+introduction of the new integration. Maintainers will review the proposal with the following checklist in mind, try and
+consider them when sharing your proposal to reduce the amount of time it takes to review your proposal. This list is not
+exhaustive, and may be updated over time.
+
+- [ ] Can the proposed component’s functionality be replicated by an existing component, with a specific configuration?
+(ex: Azure Event Hub as a `kafka` sink configuration)
+  - [ ] Alternatively implemented as a wrapper around an existing component. (ex. `axiom` wrapping `elasticsearch`)
+- [ ] Can an existing component replicate the proposed component’s functionality, with non-breaking changes?
+- [ ] Can an existing component be rewritten in a more generic fashion to cover both the existing and proposed functions?
+- [ ] Is the proposed component generically usable or is it specific to a particular service?
+  - [ ] How established is the target of the integration, what is the relative market share of the integrated service?
+- [ ] Is there sufficient demand for the component?
+  - [ ] If the integration can be served with a workaround or more generic component, how painful is this for users?
+- [ ] Is the contribution from an individual or the organization owning the integrated service? (examples of
+organization backed integrations: `databend` sink, `axiom` sink)
+  - [ ] Is the contributor committed to maintaining the integration if it is accepted?
+- [ ] What is the overall complexity of the proposed design of this integration from a technical and functional
+standpoint, and what is the expected ongoing maintenance burden?
+- [ ] How will this integration be tested and QA’d for any changes and fixes?
+  - [ ] Will we have access to an account with the service if the integration is not open source?
+- [ ] Is this integration something that can wait for a future plugin system?
```

Comment:
That's difficult to say since Vector dev team does not provide a public roadmap so external users cannot estimate - can they wait for it or not :)

Reply:
That's a fair point. @jszwedko should I just drop this from this file?

Reply:
I think we could change it to: `Would this integration be a plugin if we had a plugin system?`.

Reply:
🤔 how would an external contributor be able to discern this? Could we provide some criteria that would be examples of it being suitable for a plugin system vs not suitable ?

Reply:
Good point 😄 I think we could leave this bullet out of this doc. I think the criteria are likely to be:

- Do we expect it to be a widely used component
- Is the component implementing a generic protocol rather than a specific service integration

But we haven't fully defined them given we don't have a plugin system.

Reply:
Addressed in [bbc18b2](https://github.com/vectordotdev/vector/pull/17658/commits/bbc18b285de51e66cd01fcd1c5f89efc71bcfff1)